gateway-sp-comms/src/single_sp/update.rs

```diff
@@ -684,87 +684,25 @@ async fn send_update_in_chunks(
                 // Ideally `InvalidUpdateChunk` would return the offset the SP
                 // wants. We could add a new error variant for that; fow now,
                 // try to recover by asking the SP what chunk it expected.
-                let status = match super::rpc(
-                    cmds_tx,
-                    MgsRequest::UpdateStatus(component),
-                    None,
-                )
-                .await
-                .result
-                .and_then(expect_update_status)
-                {
-                    Ok(status) => status,
-                    Err(status_err) => {
-                        error!(
-                            log,
-                            "invalid update chunk recovery failed: \
-                             could not get update status from SP";
-                            &status_err,
-                        );
-                        return Err(err);
-                    }
-                };
-
-                // We can only recover if the SP still thinks this update is in
-                // progress.
-                let UpdateStatus::InProgress(progress) = status else {
-                    error!(
-                        log,
-                        "invalid update chunk recovery failed: \
-                         SP update status is not in progress";
-                        "status" => ?status,
-                    );
-                    return Err(err);
-                };
-
-                let UpdateInProgressStatus { id, bytes_received, total_size } =
-                    progress;
-                let id = Uuid::from(id);
-
-                if id != update_id {
-                    error!(
+                if let Some(sp_offset) =
+                    determine_update_resume_point_via_update_status(
```

Comment:
excellent function name 10/10

Reply:
I have 80 columns and I'm going to use them all gosh darn it